airbyte-commons-worker/src/main/java/io/airbyte/workers/sync/LauncherWorker.java

```diff
@@ -174,18 +186,24 @@ public OUTPUT run(final INPUT input, final Path jobRoot) throws WorkerException
         process.waitFor();
 
         if (cancelled.get()) {
-          throw new CancellationException();
+          final CancellationException e = new CancellationException();
+          ApmTraceUtils.addExceptionToTrace(e);
+          throw e;
         }
 
         final int asyncProcessExitValue = process.exitValue();
         if (asyncProcessExitValue != 0) {
-          throw new WorkerException("Orchestrator process exited with non-zero exit code: " + asyncProcessExitValue);
+          final WorkerException e = new WorkerException("Orchestrator process exited with non-zero exit code: " + asyncProcessExitValue);
+          ApmTraceUtils.addTagsToTrace(Map.of(PROCESS_EXIT_VALUE_KEY, asyncProcessExitValue));
+          ApmTraceUtils.addExceptionToTrace(e);
+          throw e;
         }
 
         final var output = process.getOutput();
 
         return output.map(s -> Jsons.deserialize(s, outputClass)).orElse(null);
       } catch (final Exception e) {
+        ApmTraceUtils.addExceptionToTrace(e);
```

Comment:
do we want to consider this an exception? It looks like if someone cancels a sync we get into here